### PR TITLE
rename c_utilities to rcutils

### PR DESCRIPTION
**Pull request by wjwwood**
I tried to break the commits into manageable parts, but for the reviewer, keep on the lookout for invalid replacements of `utilities` (and variations) with `rcutils`. This should be the biggest pr. The other pr's are coming next, but I think this can be reviewed in the meantime.

Connects to ros2/c_utilities#17

### CODE REVIEW

**Comment by wjwwood** [comment]
Ok, I think that's all the pr's that need to be opened. They're all in review. Starting some CI now.

**Comment by wjwwood** [comment]
CI:

- Linux:
  - [Build Status](http://ci.ros2.org/job/ci_linux/2491/)
- Linux-aarch64:
  - [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/90/)
- macOS:
  - [Build Status](http://ci.ros2.org/job/ci_osx/1957/)
- Windows:
  - [Build Status](http://ci.ros2.org/job/ci_windows/2601/) (known flaky tests)

**Comment by Karsten1987** [comment]
looks good to me.
Once you merged it, i guess you'll rename the git repo as well ?

File: include/rcutils/split.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
```
Comment (Karsten1987) [inline include/rcutils/split.h]: you couldn't rename this file?

Comment (wjwwood) [inline include/rcutils/split.h]: What would I rename it to?

**Comment by Karsten1987** [comment]
was more like a git thing. I've noticed that all files were actually
renamed. Splt.h however was deleted and newly added.

**Comment by wjwwood** [comment]
It was renamed: https://github.com/ros2/c_utilities/pull/19/commits/7969dd01450cfe0663f253b3d79499ab1f69222a#diff-488692e852429c0dc6e12bf8d67ed715

You're probably looking at the combined view.

**Comment by wjwwood** [comment]
Also, I'll do a merge on this pr rather than a squash, since I carefully crafted the commits.

**Comment by Karsten1987** [comment]
strange. Anyway, looks good 👍

**Pull request merged**